modules/bidwatchAnalyticsAdapter.js

```diff
@@ -22,53 +26,108 @@ function getAdapterNameForAlias(aliasName) {
+  if (typeof arg['creativeId'] == 'number') {
+    arg['creativeId'] = arg['creativeId'].toString();
+  }
+  if (removead && typeof arg['ad'] != 'undefined') {
+    arg['ad'] = 'emptied';
+  }
+  if (typeof arg['gdprConsent'] != 'undefined' && typeof arg['gdprConsent']['vendorData'] != 'undefined') {
+    arg['gdprConsent']['vendorData'] = 'emptied';
+  }
+  return arg;
+}
+
+function cleanArgs(arg, removead) {
   Object.keys(arg).forEach(key => {
-    arg[key]['originalBidder'] = getAdapterNameForAlias(arg[key]['bidderCode']);
-    if (typeof arg[key]['creativeId'] == 'number') { arg[key]['creativeId'] = arg[key]['creativeId'].toString(); }
+    arg[key] = cleanArgObject(arg[key], removead);
   });
   return arg
 }
 
-function checkBidderCode(args) {
+function checkBidderCode(args, removead) {
   if (typeof args == 'object') {
     for (let i = 0; i < objectToSearchForBidderCode.length; i++) {
-      if (typeof args[objectToSearchForBidderCode[i]] == 'object') { args[objectToSearchForBidderCode[i]] = setOriginalBidder(args[objectToSearchForBidderCode[i]]) }
+      if (typeof args[objectToSearchForBidderCode[i]] == 'object') { args[objectToSearchForBidderCode[i]] = cleanArgs(args[objectToSearchForBidderCode[i]], removead) }
     }
   }
   if (typeof args['bidderCode'] == 'string') { args['originalBidder'] = getAdapterNameForAlias(args['bidderCode']); } else if (typeof args['bidder'] == 'string') { args['originalBidder'] = getAdapterNameForAlias(args['bidder']); }
   if (typeof args['creativeId'] == 'number') { args['creativeId'] = args['creativeId'].toString(); }
+
   return args
 }
 
 function addEvent(eventType, args) {
-  if (allEvents[eventType] == undefined) { allEvents[eventType] = [] }
-  if (eventType && args) { args = checkBidderCode(args); }
-  allEvents[eventType].push(args);
+  let argsCleaned;
+  if (eventType && args) {
+    if (allEvents[eventType] == undefined) { allEvents[eventType] = [] }
+    if (saveEvents[eventType] == undefined) { saveEvents[eventType] = [] }
+    argsCleaned = checkBidderCode(JSON.parse(JSON.stringify(args)), false);
+    allEvents[eventType].push(argsCleaned);
+    saveEvents[eventType].push(argsCleaned);
+    argsCleaned = checkBidderCode(JSON.parse(JSON.stringify(args)), true);
+    if (['auctionend', 'bidtimeout'].includes(eventType.toLowerCase())) {
+      if (auctionEnd[eventType] == undefined) { auctionEnd[eventType] = [] }
+      auctionEnd[eventType].push(argsCleaned);
+    }
+  }
 }
 
 function handleBidWon(args) {
-  if (typeof allEvents.bidRequested == 'object' && allEvents.bidRequested.length > 0 && allEvents.bidRequested[0].gdprConsent) { args.gdpr = allEvents.bidRequested[0].gdprConsent; }
+  args = cleanArgObject(JSON.parse(JSON.stringify(args)), true);
+  let increment = args['cpm'];
+  if (typeof saveEvents['auctionEnd'] == 'object') {
+    for (let i = 0; i < saveEvents['auctionEnd'].length; i++) {
+      let tmpAuction = saveEvents['auctionEnd'][i];
```

Comment:
these types of loops would look a bit cleaner as 

```
for (let tmpAuction of saveEvents['auctionEnd']) { ... }`
```

or
 
```
saveEvents['auctionEnd'].forEach((tmpAuction) => {...})
```

Although I think `auction`, `bid` would be better names than `tmpAuction`, `tmpBid` etc.